src/main/java/com/github/alisonli/historyplugin/actions/SimplifyHistoryAction.java

```diff
@@ -40,39 +52,96 @@ public void update(@NotNull AnActionEvent e) {
+//            filters = logUi.getFilterUi().getFilters();
+//        }
+//        else {
+//            filters = VcsLogFilterObject.collection();
+//        }
+//
+//        MainVcsLogUi ui = VcsProjectLog.getInstance(project).openLogTab(filters);
+//        ui.getChangesBrowser().add(new JLabel("ldaf"));
 
-        List<VcsFileRevision> revisionList = diffAnalyzerService.getFileHistoryRevisionList(vcsKey, filePath);
-        List<List<Change>> changesList = new ArrayList<>();
-        if (revisionList != null && revisionList.size() > 1) {
-            try {
-                for (int i = 0; i < revisionList.size() - 1; i++) {
-                    // Assume the VCS is git
-                    GitFileRevision rev1 = (GitFileRevision) revisionList.get(i);
-                    GitFileRevision rev2 = (GitFileRevision) revisionList.get(i + 1);
-                    List<Change> changes = diffAnalyzerService.getChangesBetweenRevisions(filePath, rev1, rev2);
-                    changesList.add(changes);
-                }
-            } catch (VcsException ex) {
-                ex.printStackTrace();
-            }
+        // Version 2 Working with James to get file history
+        List<FilePath> selectedFiles = VcsContextUtil.selectedFilePaths(e.getDataContext());
+        showFileHistory(project, selectedFiles, null);
+    }
+
+    private void showFileHistory(Project project, Collection<FilePath> paths, @Nullable String revisionNumber) {
+        if (paths.size() != 1) return;
+
+        List<FilePath> pathList = new ArrayList<>(paths);
+
+        VirtualFile root = VcsLogUtil.getActualRoot(project, pathList.get(0));
+        FilePath path = getCorrectedPath(project, pathList.get(0), root, revisionNumber != null);
+        if (path.isDirectory()) return;
+
+        Hash hash = revisionNumber == null ? null : HashImpl.build(revisionNumber);
+
+        VcsLogManager logManager = VcsProjectLog.getInstance(project).getLogManager();
+        Condition<FileHistoryUi> cond = (FileHistoryUi ui) -> ui.matches(path, hash);
+        FileHistoryUi fileHistoryUi = VcsLogContentUtil.findAndSelect(project, FileHistoryUi.class,  cond);
+
+        String suffix = hash != null ? " (" + hash.toShortString() + ")" : "";
+        Function<FileHistoryUi, String> fun = (__) -> path.getName() + suffix;
+        ToolWindow toolWindow = ToolWindowManager.getInstance(project).getToolWindow(ChangesViewContentManager.TOOLWINDOW_ID);
+        if (fileHistoryUi != null) {
+            // TODO: for now, I'm passing in the same `fileHistoryUi`. You'll likely need to create a component/modify it to your needs.
+            ContentUtilEx.addTabbedContent(
+                    toolWindow.getContentManager(),
+                    tabGroupId,
+                    new TabDescriptor(new VcsLogPanel(logManager, fileHistoryUi),() -> fun.apply(fileHistoryUi), fileHistoryUi),
+                    true
+            );

```

Comment:
This is what you needed. Previously, when we were creating the factory, we were passing in the same path as what would've been passed in when the regular file history tab is created.

This is just the part of the code in the factory that creates the tab, without anything else.